src/pages/iou/request/step/IOURequestStepConfirmation.tsx

```diff
@@ -581,7 +583,12 @@ function IOURequestStepConfirmation({
 
     const onConfirm = (listOfParticipants: Participant[]) => {
         setSelectedParticipantList(listOfParticipants);
-        if (gpsRequired) {
+        const shouldStartLocationPermissionFlow =
+            !lastLocationPermissionPrompt ||
+            (DateUtils.isValidDateString(lastLocationPermissionPrompt ?? '') && DateUtils.getDifferenceInDaysFromNow(new Date(lastLocationPermissionPrompt ?? '')) > 7);
+
+        if (gpsRequired && shouldStartLocationPermissionFlow) {
```

Comment:
We could return early if `gpsRequired` is false on line 584

Reply:
I just meant that if `gpsRequired` is fale, then we don't need to check for the permission timestamp. So earlier in the function could we do this, to make the logic a bit simpler to understand?

```
if (!gpsRequired) {
    return;
}
```